oap-server/server-library/library-client/src/main/java/org/apache/skywalking/oap/server/library/client/elasticsearch/ElasticSearchClient.java

```
@@ -292,7 +300,14 @@ public int delete(String indexName, String timeBucketColumnName, long endTimeBuc
 
     public String formatIndexName(String indexName) {
         if (StringUtils.isNotEmpty(namespace)) {
-            return namespace + "_" + indexName;
+            return namespacePrefix + indexName;
+        }
+        return indexName;
+    }
+
+    public String undoFormatIndexName(String indexName) {
+        if (StringUtils.isNotEmpty(namespace) && indexName.startsWith(namespacePrefix)) {
```

If `!startWith(namespace`, then you should raise an exception.